src/test/java/uk/gov/hmcts/ccd/data/ReferenceDataRepositoryIT.java

```diff
@@ -0,0 +1,295 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
+import static uk.gov.hmcts.ccd.data.ReferenceDataRepository.BUILDING_LOCATIONS_PATH;
+import static uk.gov.hmcts.ccd.data.ReferenceDataRepository.SERVICES_PATH;
+
+@TestPropertySource(locations = "classpath:cache-refresh-schedule.properties")
+class ReferenceDataRepositoryIT extends WireMockBaseTest {
+
+    private static final String BEARER = "Bearer ";
+    private static final String SERVICE_AUTHORIZATION = "serviceauthorization";
+    private static final String TOKEN = "eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJjY2RfZ3ciLCJleHAiOjE1ODI2MDAyMzN9"
+        + ".Lz467pTdzRF0MGQye8QDzoLLY_cxk79ZB3OOYdOR-0PGYK5sVay4lxOvhIa-1VnfizaaDDZUwmPdMwQOUBfpBQ";
+    private static final String SERVICE_AUTHORISATION_VALUE = BEARER + TOKEN;
+
+    private final List<BuildingLocation> initialBuildingLocations = buildingLocations("1");
+    private final List<BuildingLocation> updatedBuildingLocations = buildingLocations("2");
+    private final List<Service> initialServices = services(11);
+    private final List<Service> updatedServices = services(22);
+
+    @Inject
+    private ReferenceDataRepository underTest;
+
+    @BeforeEach
+    void clearCache() {
+        List.of("buildingLocations", "orgServices")
+            .parallelStream()
+            .forEach(cacheName -> underTest.clearCache(cacheName));
+    }
+
+    @Test
+    void testShouldGetBuildingLocationsSuccessfullyFromUpstream() {
+        // GIVEN
+        final List<BuildingLocation> expectedBuildingLocations = buildBuildingLocations();
+        stubSuccess(BUILDING_LOCATIONS_PATH, objectToJsonString(expectedBuildingLocations));
+
+        // WHEN
+        final List<BuildingLocation> actualBuildingLocations = underTest.getBuildingLocations();
+
+        // THEN
+        assertThat(actualBuildingLocations)
+            .isNotEmpty()
+            .hasSameElementsAs(expectedBuildingLocations);
+    }
+
+    @Test
+    void testShouldGetServicesSuccessfullyFromUpstream() {
+        // GIVEN
+        final List<Service> expectedServices = buildServices();
+        stubSuccess(SERVICES_PATH, objectToJsonString(expectedServices));
+
+        // WHEN
+        final List<Service> actualServices = underTest.getServices();
+
+        // THEN
+        assertThat(actualServices)
+            .isNotEmpty()
+            .hasSameElementsAs(expectedServices);
+    }
+
+    @Test
+    void testShouldGetBuildingLocationsSuccessfullyFromCache() {
+        // GIVEN
+        stubSuccess(BUILDING_LOCATIONS_PATH, objectToJsonString(initialBuildingLocations));
+        final List<BuildingLocation> cachedBuildingLocations = underTest.getBuildingLocations();
+        removeUpstreamReferenceData();
+
+        // WHEN
+        final List<BuildingLocation> actualBuildingLocations = underTest.getBuildingLocations();
+
+        // THEN
+        assertThat(actualBuildingLocations)
```

Comment:
Could verify that underTest.getBuildingLocations() is only called once ?

Reply:
Ah right, ok  - I hadn't spotted that

Reply:
You can inject @MockBean RestTemplate or better verify wireMock call not triggered for reference_data

Reply:
Right, for this test I actually change the mapped wiremock data before making the call but, yes, I agree it would be clearer to verify no request actually got to wiremock.
I'll update the test